### PR TITLE
"snapshot-enabled" config for wan replication

### PR DESCRIPTION
**Pull request by emrahkocaman**

### CODE REVIEW

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/fullcacheconfig-applicationContext-hazelcast.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
                 <hz:property name="hazelcast.merge.first.run.delay.seconds">5</hz:property>
                 <hz:property name="hazelcast.merge.next.run.delay.seconds">5</hz:property>
             </hz:properties>
-            <hz:wan-replication name="testWan">
+            <hz:wan-replication name="testWan" snapshot-enabled="false">
                 <hz:target-cluster group-name="tokyo" group-password="tokyo-pass">
                     <hz:replication-impl>com.hazelcast.wan.impl.WanNoDelayReplication</hz:replication-impl>
```
Comment (mesutcelik) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/fullcacheconfig-applicationContext-hazelcast.xml]: I think such config should be blocked via fail fast approach. We should not allow both OSS Publisher ```WanNoDelayReplication``` and ```snapshot-enabled``` at the same time.

File: hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java [hunk 2]
```diff
@@ -694,4 +694,52 @@ public void readMulticastConfig() {
...
+        WanTargetClusterConfig targetClusterConfig = targetClusterConfigs.get(0);
+        assertEquals("test-cluster-1", targetClusterConfig.getGroupName());
+        assertEquals("test-pass", targetClusterConfig.getGroupPassword());
+        List<String> targetEndpoints = targetClusterConfig.getEndpoints();
+        assertEquals(2, targetEndpoints.size());
+        assertTrue(targetEndpoints.contains("20.30.40.50:5701"));
+        assertTrue(targetEndpoints.contains("20.30.40.50:5702"));
+    }
+
+    @Test(expected = InvalidConfigurationException.class)
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java]: can you move this test InvalidConfigurationTest class.

File: hazelcast/src/main/java/com/hazelcast/config/WanReplicationConfig.java [hunk 3]
```diff
@@ -51,11 +57,20 @@ public WanReplicationConfig setName(String name) {
...
+    public void setSnapshotEnabled(boolean snapshotEnabled) {
+        this.snapshotEnabled = snapshotEnabled;
+    }
+
     @Override
     public String toString() {
         final StringBuilder sb = new StringBuilder();
         sb.append("WanReplicationConfig");
         sb.append("{name='").append(name).append('\'');
+        sb.append(", snapshotEnabled='").append(snapshotEnabled);
```
Comment (bilalyasar) [inline hazelcast/src/main/java/com/hazelcast/config/WanReplicationConfig.java]: this single quote is not necessary after **snapshotEnabled**

File: hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java (hunk 2 above)
Comment (mesutcelik) [inline hazelcast/src/test/java/com/hazelcast/config/XMLConfigBuilderTest.java]: good to see that we have fail fast test cases in config related PRs.
:+1:

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.5.xsd [hunk 4]
```diff
@@ -1533,6 +1533,11 @@
                 <xs:restriction base="xs:string"/>
             </xs:simpleType>
         </xs:attribute>
+        <xs:attribute name="snapshot-enabled" use="optional">
```
Comment (mesutcelik) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.5.xsd]: default = false ?

**Comment by mesutcelik** [comment]
needs a rebase

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/fullcacheconfig-applicationContext-hazelcast.xml (hunk 1 above)
Comment (emrahkocaman) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/fullcacheconfig-applicationContext-hazelcast.xml]: Fail-fast check added.

**Comment by emrahkocaman** [comment]
Rebased.

**Pull request merged**